File: PawNClaw.Backend/PawNClaw.Data/Database/CageType.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

#nullable disable

namespace PawNClaw.Data.Database
{
    [Index(nameof(Id), Name = "i")]
    public partial class CageType
    {
        public CageType()
        {
            Cages = new HashSet<Cage>();
            FoodSchedules = new HashSet<FoodSchedule>();
            Prices = new HashSet<Price>();
        }

        [Key]
        [Column("id")]
        public int Id { get; set; }
        [Required]
        [Column("type_name")]
        [StringLength(256)]
        public string TypeName { get; set; }
        [Column("description")]
        [StringLength(512)]
        public string Description { get; set; }
        [Column("height", TypeName = "numeric(19, 5)")]
        public decimal Height { get; set; }
        [Column("width", TypeName = "numeric(19, 5)")]
        public decimal Width { get; set; }
        [Column("length", TypeName = "numeric(19, 5)")]
        public decimal Length { get; set; }
        [Column("is_single")]
        public bool IsSingle { get; set; }
        [Column("create_date", TypeName = "date")]
        public DateTime? CreateDate { get; set; }
        [Column("modify_date", TypeName = "date")]
        public DateTime? ModifyDate { get; set; }
        [Column("create_user")]
        public int? CreateUser { get; set; }
        [Column("modify_user")]
        public int? ModifyUser { get; set; }
        [Column("status")]
        public bool? Status { get; set; }
        [Column("center_id")]
        public int CenterId { get; set; }

        [NotMapped]
        public string TotalPrice { get; set; }

        [NotMapped]
        public bool? CanDelete { get; set; }

        [ForeignKey(nameof(CenterId))]
        [InverseProperty(nameof(PetCenter.CageTypes))]
        public virtual PetCenter Center { get; set; }
        [ForeignKey(nameof(CreateUser))]
        [InverseProperty(nameof(Account.CageTypeCreateUserNavigations))]
        public virtual Account CreateUserNavigation { get; set; }
        [ForeignKey(nameof(ModifyUser))]
        [InverseProperty(nameof(Account.CageTypeModifyUserNavigations))]
        public virtual Account ModifyUserNavigation { get; set; }
        [InverseProperty(nameof(Cage.CageType))]
        public virtual ICollection<Cage> Cages { get; set; }
        [InverseProperty(nameof(FoodSchedule.CageType))]
        public virtual ICollection<FoodSchedule> FoodSchedules { get; set; }
        [InverseProperty(nameof(Price.CageType))]
        public virtual ICollection<Price> Prices { get; set; }

        [NotMapped]
        public ICollection<Photo> Photos { get; set; }

        [NotMapped]
        public decimal MaxPrice { get; internal set; }

        [NotMapped]
        public decimal MinPrice { get; internal set; }
    }
}

File: PawNClaw.Mobile/lib/models/cage_type.dart
import 'package:pawnclaw_mobile_application/models/photo.dart';

import 'cage.dart';

class CageTypes {
  int? id;
  String? typeName;
  String? description;
  double? height;
  double? width;
  double? length;
  bool? isSingle;
  List<Cages>? cages;
  double? totalPrice;
  double? minPrice;
  double? maxPrice;
  Photo? photo;

  CageTypes(
      {this.id,
      this.typeName,
      this.description,
      this.height,
      this.width,
      this.length,
      this.isSingle,
      this.cages,
      this.totalPrice,
      this.minPrice,
      this.maxPrice,
      this.photo});

  CageTypes.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    typeName = json['typeName'];
    description = json['description'];
    height = json['height'];
    width = json['width'];
    length = json['length'];
    isSingle = json['isSingle'];
    if (json['cages'] != null) {
      cages = <Cages>[];
      json['cages'].forEach((v) {
        cages!.add(new Cages.fromJson(v));
      });
    }
    totalPrice =
        json['totalPrice'] == null ? null : double.parse(json['totalPrice']);
    minPrice = json['minPrice'] == null ? null : json['minPrice'].toDouble();
    maxPrice = json['maxPrice'] == null ? null : json['maxPrice'].toDouble();

    // if (json['prices'] != null) {
    //   prices = <Null>[];
    //   json['prices'].forEach((v) {
    //     prices!.add(new Null.fromJson(v));
    //   });
    // }
    if (json['photos'] != null) {
      var photos = <Photo>[];
      json['photos'].forEach((v) {
        photos.add(Photo.fromJson(v));
      });
      if(photos.isNotEmpty) photo = photos.first;
    }
  }
  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['typeName'] = this.typeName;
    data['description'] = this.description;
    data['height'] = this.height;
    data['width'] = this.width;
    data['length'] = this.length;
    data['isSingle'] = this.isSingle;
    if (this.cages != null) {
      data['cages'] = this.cages!.map((v) => v.toJson()).toList();
    }
    if (this.totalPrice != null) data['totalPrice'] = this.totalPrice.toString();
    if (this.maxPrice != null) data['maxPrice'] = this.maxPrice.toString();
    if (this.minPrice != null) data['minPrice'] = this.minPrice.toString();
    if (this.photo != null) {
      data['photos'] = this.photo!.toJson();
    }
    return data;
  }
}
